mangaki/mangaki/models.py

```diff
@@ -248,6 +248,31 @@ def update_tags(self, anidb_tags):
 
     # Should add an update_creators method similar to update_tags
 
+    def is_nsfw_based_on_tags(self, anidb_tags):
+        # Might want to move this somewhere else ...
+        potentially_nsfw_tags = ['nudity', 'ecchi', 'pantsu', 'breasts', 'sex',
+                                 'large breasts', 'small breasts', 'gigantic breasts',
+                                 'incest', 'pornography', 'shota', 'masturbation',
+                                 'sexual fantasies', 'anal', 'loli']
+
+        if self.nsfw:
+            return
+
+        sum_weight_all_low = sum([infos["weight"] for t, infos in anidb_tags.items()
+                                  if infos["weight"] <= 400])
+        sum_weight_nsfw_low = sum([infos["weight"] for t, infos in anidb_tags.items()
+                                   if t in potentially_nsfw_tags and infos["weight"] <= 400])
+
+        sum_weight_all_high = sum([infos["weight"] for t, infos in anidb_tags.items()
+                                   if infos["weight"] > 400])
+        sum_weight_nsfw_high = sum([infos["weight"] for t, infos in anidb_tags.items()
+                                    if t in potentially_nsfw_tags and infos["weight"] > 400])
+
+        percent_low_nsfw = (sum_weight_nsfw_low/sum_weight_all_low) * 100
+        percent_high_nsfw = (sum_weight_nsfw_high/sum_weight_all_high) * 100
+
+        self.nsfw = (percent_high_nsfw > 15 or percent_low_nsfw > 30)
```

Comment:
Add a `# FIXME: 15 and 30 should be configurable constants`
Let's call them `HIGH_NSFW_THRESHOLD` and `LOW_NSFW_THRESHOLD` if you have time.